homeassistant/components/elmax/config_flow.py

```diff
@@ -0,0 +1,249 @@
+        collisions_count += 1
+        panel_name = f"{original_panel_name} ({collisions_count})"
+    panel_names[panel_name] = panel_id
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for elmax-cloud."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize."""
+        self._client: Elmax = None
+        self._username: str = None
+        self._password: str = None
+        self._panels_schema = None
+        self._panel_names = None
+        self._reauth_username = None
+        self._reauth_panelid = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        # When invokes without parameters, show the login form.
+        if user_input is None:
+            return self.async_show_form(step_id="user", data_schema=LOGIN_FORM_SCHEMA)
+
+        errors: dict[str, str] = {}
+        username = user_input[CONF_ELMAX_USERNAME]
+        password = user_input[CONF_ELMAX_PASSWORD]
+
+        # Otherwise, it means we are handling now the "submission" of the user form.
+        # In this case, let's try to log in to the Elmax cloud and retrieve the available panels.
+        try:
+            client = Elmax(username=username, password=password)
+            await client.login()
+
+            # If the login succeeded, retrieve the list of available panels and filter the online ones
+            online_panels = [x for x in await client.list_control_panels() if x.online]
+
+            # If no online panel was found, we display an error in the next UI.
+            panels = list(online_panels)
+            if len(panels) < 1:
+                raise NoOnlinePanelsError()
+
+            # Show the panel selection.
+            # We want the user to choose the panel using the associated name, we set up a mapping
+            # dictionary to handle that case.
+            panel_names: dict[str, str] = {}
+            username = client.get_authenticated_username()
+            for panel in panels:
+                _store_panel_by_name(
+                    panel=panel, username=username, panel_names=panel_names
+                )
+
+            self._client = client
+            self._panel_names = panel_names
+            schema = vol.Schema(
+                {
+                    vol.Required(CONF_ELMAX_PANEL_NAME): vol.In(
+                        self._panel_names.keys()
+                    ),
+                    vol.Required(CONF_ELMAX_PANEL_PIN, default="000000"): str,
+                }
+            )
+            self._panels_schema = schema
+            self._username = username
+            self._password = password
+            return self.async_show_form(
+                step_id="panels", data_schema=schema, errors=errors
+            )
+
+        except ElmaxBadLoginError:
+            _LOGGER.error("Wrong credentials or failed login")
```

Comment:
Do we need to log this as an error; while also already providing feedback in the UI?